Added home_mode

Duoxilian opened the pull request:
**Description:**

As discussed in issue #4950, the ecobee (and climate) components implement only an away_mode hold. However, the home_mode hold is critical for ecobee automation. This feature adds this mode; it is modelled after away_mode.

**Related issue (if applicable):** fixes #4950, #4953, In addition, the temperature attribute is provided which gives the current target temperature (instead of having to determine that from the low and high targets). This is more consistent with ecobee also.

**Pull request in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io) with documentation (if applicable):** home-assistant/home-assistant.github.io#1700

**Checklist:**

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io)

If the code communicates with devices, web services, or third-party tools:
  - [x] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
  - [x] New dependencies have been added to the `REQUIREMENTS` variable ([example][ex-requir]).
  - [x] New dependencies are only imported inside functions that use them ([example][ex-import]).
  - [x] New dependencies have been added to `requirements_all.txt` by running `script/gen_requirements_all.py`.
  - [x] New files were added to `.coveragerc`.

If the code does not interact with devices:
  - [ ] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
  - [ ] Tests have been added to verify that the new code works.

[ex-requir]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L16
[ex-import]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L51

balloob committed:
Version bump to 0.36.0.dev0

balloob committed:
Gracefully exit with async logger (#4965)

* Gracefully exit with async logger

* Lint

fabaff committed:
Remove and update docstrings (#4969)

titilambert committed:
Add HydroQuebec support (#4840)

quadportnick committed:
Add support for the Sonarr URL Base setting (#4975)

* Add support for the Sonarr URL Base setting

For those of us who have sonarr hidden behind reverse proxy under a path, we need to be able to pass the path

Adds urlbase: XXX to the sonarr yaml... Match it to what you have set in Sonarr (Settings>General>URL Base)

* Fix line lengths

* Fix trailing white space caused by last change

* Removing use of len()

Duoxilian committed:
Added home_mode

Duoxilian committed:
Added home_mode

pvizeli committed:
Bugfix TTS clear cache (#4974)

* Bugfix TTS base url with certificate

* fix lint

* remove base_url stuff fix only clear_cache stuff

* cleanup

balloob committed:
Allow setting base url (#4985)

pvizeli committed:
Bugfix wait in automation (#4984)

andrey-git committed:
Fix non-radio Sonos album-art by using track_info['album_art'] before checking other options. (#4958)

* Fix Sonos album art for non-radio streams

* Revert "Fix Sonos album art for non-radio streams"

This reverts commit d71502d18f5778146cf45dcb717aba71f88a3fac.

* Fix Sonos album art for non-radio streams

* Move art existance check into _format_media_image_url

balloob committed:
Base url: Fix external port different from internal port (#4990)

* Base url: Fix external port different from internal port

* Add base_url example to new config

balloob committed:
Merge branch 'master' into dev

balloob committed:
Unbreak dev

Danielhiversen committed:
Support for broadlink sp (#4961)

* initial support for broadlink sp

* style fix

* style

* bug fix

MartinHjelmare commented on homeassistant/components/climate/__init__.py:
```diff
@@ -219,6 +237,32 @@ def away_mode_set_service(service):
+
+        home_mode = service.data.get(ATTR_HOME_MODE)
+
+        if home_mode is None:
```
This will never happen since the schema will validate the passed attributes before this function is called. You can remove this whole block.

Duoxilian replied:
Regarding the "this will never happen". The same check are already in the code for "away_mode_set_service", so for consistency I would like retain it. Also, I found that services crash when one is missing parameters so I am not sure about your comment regarding "this will never happen". See issue #4871.

Duoxilian replied:
OK. Removed the block. One should update all the other similar functions to remove this there as well, for consistency reasons. But I'd rather have another PR for this.

Duoxilian replied:
After I removed the block per your instructions, the test for wrong parameters no longer passes:

self = <tests.components.climate.test_demo.TestDemoClimate testMethod=test_set_home_mode_bad_attr>

    def test_set_home_mode_bad_attr(self):
        """Test setting the home mode without required attribute."""
        state = self.hass.states.get(ENTITY_CLIMATE)
>       self.assertEqual('on', state.attributes.get('home_mode'))
E       AssertionError: 'on' != None

I put the block back in, as it is in all other services in the climate component. If you don't agree, I suggest that there is a greater systematic discussion about what the climate component should do, not just my small additional method.

Duoxilian replied:
OK. Updated the tests; realized I had missed some initialization. Took the block out as you requested (albeit inconsistent with the rest).

MartinHjelmare reviewed (changes requested):
In general looks good, with one comment, although I'm not familiar with the ecobee API. Couple of lint comments to fix too.

MartinHjelmare commented:
You should also update the demo climate platform and tests for the new service, method and attribute.

Commit:
rfxtrx lib upgrade

Danielhiversen committed:
Merge pull request #5000 from home-assistant/rfxtrx_lib_0.14

rfxtrx lib upgrade

jabesq committed:
Hotfix for Netatmo Camera (#4998)

Signed-off-by: Hugo D. (jabesq) <jabesq@gmail.com>

Duoxilian committed:
Correct lint issues.

Duoxilian committed:
Correct lint issues.

Duoxilian committed:
Merge branch 'master' into dev

Duoxilian committed:
Added tests for home_mode

Duoxilian committed:
Added tests for home_mode

Duoxilian committed:
Merge branch 'dev' of https://github.com/duoxilian/home-assistant into dev

Duoxilian committed:
Correct lint issues and add tests.

Duoxilian committed:
Correct lint issues and add tests.

danielperna84 committed:
Add missing support for HMIP-PSM (#5013)

konikvranik committed:
script/lint only on python files (#5018)

Danielhiversen committed:
Solve some bugs in the bradlink switch

Danielhiversen committed:
style fix

Danielhiversen committed:
Merge pull request #5022 from home-assistant/broadlink_bug_fix

Solve some bugs in the broadlink switch

kellerza committed:
device_tracker (#5023)

2

kellerza committed:
Spread seconds (#5025)

pvizeli committed:
Bugfix voicerss post api (#5021)

* Bugfix voicerss post api

* fix unittest

* Add cache to service description

pvizeli committed:
Fix spell media_player service (#5030)

Add an optional extended description…

pvizeli committed:
Bugfix create a task from a task in component update (#5033)

Duoxilian committed:
Respond to Martin's request.

Duoxilian committed:
Respond to Martin's request.

Duoxilian committed:
Undo of Martin's request due to test failure.

Duoxilian committed:
Undo of Martin's request due to test failure.

Duoxilian committed:
Fixed test failures and included Martin's suggestion.

Duoxilian committed:
Fixed test failures and included Martin's suggestion.

pvizeli committed:
Bugfix async log handle re-close bug (#5034)

* Bugfix async log handle re-close bug

* Check on running thread on async_close

* Fix now on right place

mezz64 committed:
Update pyHik requirement version (#5040)

technicalpickles committed:
Nest fixes (#5011)

* Updated Nest API to have logical names

* Fix NoneType not having replace method in NestSensor constructor

* Move name setting to constructor, in case zone.name causes IO.

* normalize is_online to online

* Updated python-nest API

* push is_* helpers down to python-nest, and use inheritence to implement rather than checking class name

* Update python-nest

Duoxilian commented:
Martin, could you please approve? All your requests have been satisfied. I have removed the code block you questioned, and I have added relevant handling in the demo climate platform.

Commit:
Merge branch 'master' into dev

Duoxilian committed:
Merge branch 'master' into dev

MartinHjelmare reviewed (approved):
Great! Now you just need to update the docs with the new service:
https://home-assistant.io/components/climate/

Duoxilian committed:
Merge branch 'master' into dev

Duoxilian commented:
Martin, I am embarassed to say but I cannot figure out how to update the documentation. I did the "edit this file on github" thing, and this seems to force me to create a fork. I then edited the documentation file, but now I don't see how I can create the pull request there. When I try to create a pull request it tells me I have nothing changed. But the fork page clearly shows the differences. Any guidance or pointers?

Duoxilian commented:
Well, I succeeded in making a pull request, but I don't know whether this was correct as it does not show any of my changes; well, we shall see...

MartinHjelmare commented:
https://github.com/home-assistant/home-assistant.github.io/pull/1652

abmantis committed:
emulated_hue: fix alexa "device not responding" (#5058)

* emulated_hue: fix alexa "device not responding"

we need to set the brightness to 100 for devices that only turn on

* emulated_hue: dont override brightness for scenes/scripts

* emulated_hue: python and semi-colons

* emulated_hue: fix output brightness level

abmantis committed:
Emulated hue: add support for cover open/close (#5057)

* add support for cover open/close

* fix action compare; reduce line width

aequitas committed:
Fix and test for prefixed MAC addresses. (#5052)

* Fix and test for prefixed MAC addresses.

* Fix style.

* Don't commit old code.

* Fix style.

pvizeli committed:
Async migrate climate (#5026)

* Async migrate climate

* Change update handling

pvizeli committed:
Async reduce coro (#5001)

* Asyncio coro reduce

* Replace comments

HydrelioxGitHub committed:
Add sensor for International Space Station (#4968)

* Add sensor for International Space Station

* Change two sensors to one with attributes.

* Fix due to comments in HA PR. Thanks !

* Update Requirement

fabaff committed:
Upgrade pyowm to 2.6.0 (#5071)

fabaff committed:
Upgrade psutil to 5.0.1 (#5072)

fabaff committed:
Upgrade speedtest-cli to 1.0.1 (#5073)

Danielhiversen committed:
upgrade miflora (#5075)

Add an optional extended description…

andrey-git committed:
Allow to specify TTS language in the service call. (#5047)

* Allow to specify TTS language in the service call.

* Allow to specify TTS language in the service call.

* Respect 79 char limit

* Fix "Too many blank lines"

* Fix "Too many blank lines"

* Fix "Too many blank lines"

* Change language to be optional parameter of *get_tts_audio

* Change language to be optional parameter of *get_tts_audio

* Respect 79 char limit

* Don't pass "None

* Use default of "None" for TTS language

* Use default of "None" for TTS language

* Don't pass "None"

* Change TTS cache key to be hash_lang_engine

* Change language from demo to en

* Fix wrong replace

michaelarnauts committed:
Update icons from materialdesignicons.com (#5081)

balloob commented:
How is enabling home mode different from disabling away mode?

I also think that adding another service to the climate component is not a scalable solution, instead we should probably aim for a `set_mode` service.

btorresgil committed:
Ecobee service fix and new 'resume program' service (#4510)

* ecobee_set_fan_min_on_time: fix issue using 'entity_id' field and add service field help text

* climate.ecobee: add 'resume_program' service

* Add default value for resume_all and correct entity_id field name reference

turbokongen committed:
Improvements to zwave lock platform (#5066)

magicus committed:
Include flake8-docstrings to test requirements to better mimic tox -e lint (#4926)

magicus committed:
Fix typo. (#5087)

kellerza committed:
Fix up docstring for tests (#5090)

Commit:
Added Ledenet protocol support to flux_led (#5097)

* Added Ledenet protocol support to flux_led

* Made the protocol config lowercase

kellerza committed:
Qwikswitch library PyPi update (#5099)

gopalkildoliya committed:
Facebook Notify Started

Commit:
Replace dots in Alexa built-in intent slots w/ underscores (#5092)

* Replace dots in built-in intent slots w/ underscores

* Add a built-in intent test

pvizeli committed:
Bugfix close async log handler (#5086)

titilambert committed:
Fix proximity issue (#5109)

Commit:
Fixes moldindicator sensor after switch to asyncio (#5038)

gopalkildoliya committed:
Facebook notify updated

gopalkildoliya committed:
File added to coveragerc

scmmmh committed:
Philio 3-in-1 Gen 4 zwave sensor needs the no-off-event workaround. (#5120)

andrey-git committed:
Add print_config_parameter service to Z-Wave (#5121)

* Add print_config_param service to z-wave

* Add print_config_parameter service to z-wave

* Add print_config_parameter service to z-wave

* Fix typos

* Fix typos

* Fix typo

jtscott committed:
[climate] Fix typo in services.yaml (#5136)

pavoni committed:
Bump pyloopenergy to catch SSL exception.

pavoni committed:
Merge pull request #5142 from home-assistant/bump_loop_energy

Bump pyloopenergy to catch SSL exception.

turbokongen committed:
Catch RuntimeError (#5134)

andrey-git committed:
Clean up DEFAULT_DEBUG constant in zwave (#5138)

Nice :+1:

turbokongen committed:
Prevent Homeassistant to create a segfault with OZW (#5085)

turbokongen committed:
[zwave]Use schedule_ha_state and add debug message (#5143)

* Use schedule_ha_state and add debug message

* Logger not defined

pvizeli committed:
Update device_traker for async platforms (#5102)

Async DeviceScanner object, migrate to async, cleanups

michaelarnauts committed:
Improve development workflow in docker (#5079)

* Allow bower install of frontend components as root. Needed for frontend development in docker since everything runs as root in the docker image.

* Improve development workflow in docker

* Use LANG=C.UTF-8 in tox. Fixes installation of libraries with UTF-8 in it's readme.

* Install mysqlclient psycopg2 uvloop after requirements_all.txt again, but with a --no-cache-dir this time. Allows bootstrap_frontend to be executed in a different path like the other scripts.

Duoxilian commented:
Thermostats generally have a mode, which would be the basic setting: typically "Home", "Away", "Sleep", sometimes additional ones. Thermostats have also overrides, when you can set a higher or lower temperature then programmed in the mode. And then some thermostats have special overrides, refered to as "smart home" and "smart away" or something similar. "Smart home" would apply the home settings as override, while "smart away" applies the away settings as override. 

The current climate component supports only "smart away" (it is called "away_mode" in the component). But it does not support the "smart home" mode. This request provides that additional mode. Note that "smart home" is not the same as "not smart away", it is a distinct different behavior. Of course one could also define a mode "override" with various values such as "home", "away", "off", etc. However, this would be inconsistent with the current implementation of "away_mode". Whoever implemented that mode boxed us into the style of separate modes where one override mode with multiple values would have sufficed.

While I agree with you that a muiltivalued override mode would be more scalable, I don't see a solution that is both consistent and backwards compatible. For this reason, I provided the "home_mode".

Zac-HD committed:
Fix link to pull request advice for contributors

kk7ds committed:
Nx584 maint (#5149)

* Update nx584 requirement to 0.4

There have been a few bug fixes to nx584 since 0.2, so this just updates
our requirement to pull in the newer version.

* Fix nx584 if no partitions are found

If we succeed in our connection to the panel but find no
configured partitions, then we will fall through to the bypass
probe and fail because 'zones' is never initialized. This fixes
both exception paths and adds a test that would poke it.

Duoxilian commented:
To add some more, I see two options here:

Assuming a thermostat provides overrides for the current mode to temporarily use the settings for "away", "home", "sleep", etc. (I am only aware of the first two but we can imagine that vendors get even more creative):

1. Have an attribute, say "override_mode", with various values of possible overrides provided by the thermostat, for example, "away", "home", "sleep", etc. 

2. Have multiple attributes for each override, say "away_mode", "home_mode", "sleep_mode", etc., with values "on" and "off".

Clearly (1) is more scalable, but (2) is the current implementation.

Please advise how you want to proceed in providing the override by home setting. If you do not want this capability in climate, let me know, and I move it down to ecobee. But surely other thermostats will eventually provide this, or may already do so (I am not familiar with other smart thermostats).

brandonweeks committed:
Add version test for monkey_patch_asyncio() (#5127)

* Add version test for monkey_patch_asyncio()

* Update __main__.py

emlove committed:
Merge pull request #5150 from Zac-HD/fix-contrib-link

Fix link to pull request advice for contributors

andrey-git committed:
Keep previous brightness of dimming zwave light upon turning it on/off (#5160)

* Keep previous brightness of dimming zwave light upon turning it on/off

* Fix initialization

* Use value 255 to set dimmer to previous value

Danielhiversen committed:
issue #5101 (#5161)

eieste committed:
Apple TouchBar icon Support (mask-icon) (#5159)

* Update index.html

Add Apple TouchBar icon Color Support

* Add svg icon

* Change path to SVG icon

* Rename mask-icon.svg to home-assistant-icon.svg

* Remove useless whitespace

* Update index.html

balloob committed:
Run tests on Python 3.6 (#5162)

* Run tests on Python 3.6

* Fix dsmr test

* Fix async util tests

* Fix rest sensor test

henworth committed:
Change zeroconf to use local ip not base_url (#5151)

Rather than rely on base_url being unconfigured and therefore be
set as the host ip in the api component, get the local ip directly.

Use server port from the http component instead of the api component
where it will be None if base_url is set.

Change the exception catch meant to check for ipv4 vs ipv6 to wrap the
address encoding only instead of the zeroconf service info declaration.

emlove committed:
Kodi cleanup and config update (#5098)

kellerza committed:
[mqtt] Embedded MQTT server fix (#5132)

* Embedded MQTT server fix and schema

* feedback

Commit:
Add support for limiting which entities are recorded (#4733)

brandonweeks committed:
Upgrade samsungctl to 0.6.0 (#5126)

Giannie committed:
Allow selection of bluetooth device to use (#5104)

Adding the device_id key to the bevice tracker component allows selecting the bluetooth device to use in case the default device does not have BLE Capabilities

titilambert committed:
Improve Sharp Aquos TV component - Fixes #4973 (#5108)

Danielhiversen committed:
Broadlink fix (#5065)

* Broadlink fix

* style fix

* style fix

* typo

* restructure

* Update broadlink.py

* Update broadlink.py

* Add support for more devices

* fix library version

* fix library version

* fix library version

* fix library version

* fix library version

* Update broadlink.py

* lib version

* remove lower

* remove lower

* refactor

* refactor

* refactor

* authorization

* authorization

* refactor

* lib version

* lib version

n8henrie committed:
Add ability to set rpi_rf `tx_repeats` attribute (#5070)

* Add ability to set rpi_rf `tx_repeats` attribute

Uses the current `rpi_rf` default of 10
Closes home-assistant/home-assistant#5069

* Use `signal_repetitions` instead of `repeats` for consistency

emlove committed:
Async support in media player component (#4995)

* Async support in media player component

* Removed redundant new tests

* Update to new 'reduce coroutine' standards

* Remove extra create_task

magicus committed:
Improve async/multithreaded behavior of tellstick code (#4989)

* Refactor tellstick code for increased readability. Especially highlight if "device" is a telldus core device or a HA entity.

* Refactor Tellstick object model for increased clarity.

* Update comments. Unify better with sensors. Fix typo bug. Add debug logging.

* Refactor tellstick code for increased readability. Especially highlight if "device" is a telldus core device or a HA entity.

* Refactor Tellstick object model for increased clarity.

* Update comments. Unify better with sensors. Fix typo bug. Add debug logging.

* Fix lint issues.

* Remove global variable according to hint from balloob.

* Better async/threading behavior for tellstick.

* Fix lint/style checks.

* Use hass.async_add_job

Commit:
Pushed to another version of denonavr library with fixes for AVR-nonX devices (#5156)

florianholzapfel committed:
fix #5157 (#5173)

gopalkildoliya committed:
Timeout and Constant added

kk7ds committed:
Bump uvcclient to 0.10.0 (#5175)

This brings fixes for newer versions of unifi-video and a few
fixes.

michaelarnauts committed:
Ping device tracker (#5176)

* Ping device tracker

* Style fixes

Danielhiversen committed:
user agent header (#5172)

* user agent in header

* update user agent info

* Use user-agent from lib

doudz committed:
add offline tts using pico (#5005)

* add offline tts using pico

* Update picotts.py

* Update picotts.py

* Update picotts.py

* Update picotts.py

* Update picotts.py

* Update picotts.py

* Update picotts.py

* Update .coveragerc

Commit:
Add support for Tikteck Bluetooth bulbs (#4843)

* Add support for Tikteck Bluetooth bulbs

Adds support for the Tikteck RGBW BLE bulbs. These don't provide "true" RGBW
support - at a certain point in RGB space, the white LEDs turn on. Each bulb
has a specific key that needs to be extracted from the Android app.

* Update tikteck.py

Commit:
Rewrite influxdb metrics to be more consistent (#4791)

* Updated to make all metrics consistent

* Updated existing test for new format

* Updated checks on lists and dictionarys

Danielhiversen committed:
[WIP] Spread the traffic to yr.no servers and remove yr.no from the default… (#5171)

* Spread the traffic to yr.no servers and remove yr.no from the default config

* use unique address for yr.no

* update yr tests

* Wait 10 min extra before requesting new data

* Update config.py

* Update yr.py

sdeleeuw committed:
Fixed invalid response when sending a test-request from Locative iOS app (#5179)

happyleavesaoc committed:
USPS sensor (#5180)

* usps sensor

* Update usps.py

michaelarnauts committed:
Speeds up lint and test in docker by keeping the cache between invocations. (#5177)

* Add a volume to store the tox cache on the host. This gives quite some speed boost when running lint_docker and test_docker.

* Only map .tox directory for cache.

kellerza committed:
ensure_list validator - Allow None to return an empty list (#5133)

MatoKafkac committed:
Beaglebone black GPIO control by switch v2 (#4908)

* Added support for BBB GPIO

* Requirements updated

* unnecessary pylint statement removed

* Changed according arduino switch

* typo corrected

* Hound errors solved

* lint error

* Update bbb_gpio.py

emlove committed:
Convert Kodi platform to async (#5167)

* Convert Kodi platform to async

* Remove unnecessary async_update_ha_state

MrMep committed:
Update keyboard_remote.py (#5112)

* Update keyboard_remote.py

I changed os.path.isFile() to os.path.exists: as far as I know isFile doesn't work with device files. At least on my Ubuntu it wasn't working.

Then I added some error control in case the keyboard disconnects: with bluetooth keyboards this happen often due to battery saving. Now it reconnects automatically when the keyboard wakes up.

We could fire an event to hass when the keyboard connects-disconnects, maybe I'll do this later.

We should also manage errors due to permissions problems on the device file, or at least give some info in the docs about how to allow HA to grab control over an system input file.

I'm sorry if my coding isn't up to some standard practice I'm not aware of: I'm new to HA and to python itself, I'm just trying to be of help.
Gianluca

* Update keyboard_remote.py

I changed some other few things.

Not the component gets loaded even if the keyboard is disconnected. When it connects, it starts to fire events when keys are pressed.

I also added a sleep(0.1) that reduces a lot the load on the CPU, but without many consequences on key pressed detection.

webworxshop committed:
Add support for customised Kankun SP3 Wifi switches. (#5089)

* Add support for customised Kankun SP3 Wifi switches.

* Add config validation for Kankun SP3 wifi switch.

* Update kankun.py

Commit:
Update frontend

balloob commented:
I like option (1). I think it's fine to break backwards compatibility, we need to make sure the frontend is updated too.

partofthething committed:
Support longer-than-60-second scan_interval and interval_seconds (#5147)

* Update scan_interval and interval_seconds max to 1 day vs. 60 seconds

* Format fixes

* Add docstring on unittest.

* Added and implemented new async_track_time_interval helper.

* Format fixes, removed unused import.

* Undid whoops on unsub_polling.

* Updated unit tests for scan_interval.

* Added unit test for track_time_interval.

* Allow other forms of time interval input for scan_interval and interval_seconds

pvizeli committed:
Bugfix aiohttp connector pool close on shutdown (#5190)

* Bugfix aiohttp connector pool close on shutdown

* fix circular import

* remove lint disable

emlove committed:
Fix for async device_tracker (#5192)

michaelarnauts committed:
Universal media_player returns ATTR_MEDIA_POSITION and ATTR_MEDIA_POSITION_UPDATED_AT from it's active child now. (#5184)

rmkraus committed:
Re-enabled Weather Sensors for the ISY component. (#5148)

* Re-enabled Weather Sensors for the ISY component.

As the ISY component had been updated to support newer components and
have better UOM support, the support for ISY’s weather module was
dropped. This adds that support back into Home Assistant.

* Cleanup of the ISY Weather support.

Cleaned up the for loops used to generate nodes representing weather
data from the ISY. Moved the weather_node named tuple to the top of the
file so that it can be more easily identified.

* Update isy994.py

jjensn committed:
Add Yeelight auto discovery, color temp (#5145)

* Add Yeelight auto discovery, color temp

* Fixing linting issues

pvizeli committed:
Bugfix default values to timedelta (#5193)

* Bugfix default values to timedelta

* fix unittests

pvizeli committed:
[new] component rest_command (#5055)

* New component rest_command

* add unittests

* change handling like other command

* change unittest

* address @balloob comments

balloob committed:
Fix tests closing properly (#5203)

Duoxilian commented:
@balloob, I assume you realize that if we do this every climate implementation will have to be changed, and there are 16 of them. I think this is not something we should take lightly. Thermostats are probably the most central part of most people's home automation. I would not want to rewrite thermostat code I cannot test. Maybe I should move this extra mode down to Ecobee only and not mess with other thermostats? If this issue comes up with others, we can deal with it then.

lwis committed:
[calendar] Fix scan interval (#5205)

pvizeli committed:
Improve aiohttp default clientsession close with connector (#5208)

lwis committed:
[device_tracker] Use home zone GPS coordinates for router based devices. (#4852)

pvizeli committed:
Bugfix segfault with new helper track_time_interval (#5222)

* Bugfix sigfault with new helper track_time_interval

* Add none init also to sunrise/sunset for consistance

janLo committed:
Add teardown method to pilight tests (#5195)

* Add teardown method to pilight tests

This is necessary to stop the HomeAssistant instance that was started
in the setUp method. Without this there happen random test failures.

This is necessary to stabilize the tests for PR #5045.

Signed-off-by: Jan Losinski <losinski@wh2.tu-dresden.de>

* Update test_pilight.py

dasos committed:
Squeezebox JSON-RPC (#5084)

* Refactor of Squeezebox connection code

* Refactor of Squeezebox connection code

* Typos

* Make Python 3.4 friendly

* Addressing comments

* Improving docstring

* Using discovered port

* Style better

* Accept new disco object

* Revert "Accept new disco object"

* Make it obvious that port isn't discovered yet

* Flake8. ;)

happyleavesaoc committed:
usps sensor: better delivery handling (#5202)

* better delivery handling

* bump dep version

happyleavesaoc committed:
date fix (#5227)

markferry committed:
onkyo: fix selecting sources with only one name (#5221)

craigjmidwinter committed:
Insteon local (#5088)

* platform set-up begin components

* lights seem to be getting set up properly, not sure why they aren't being added...

* typo

* Dependencies line

* toggle working

* toggle working

* added the switch to insteon_local

First commit hope to test tonight or in the morning

* Update insteon_local.py

* Update insteon_local.py

* Update insteon_local.py

* Update insteon_local.py

* Update insteon_local.py

* Update insteon_local.py

* Update insteon_local.py

* Update insteon_local.py

* Update insteon_local.py

* Update insteon_local.py

* Update insteon_local.py

* move dependency declaration before import?

* Move dependencies in Switch

* Update insteon_local.py

* wait for response

* switched the while to an if

switched the while 'cmd2' not in resp: to an if 'cmd2' not in resp: this seems to have the updater working

* Switched the while sleep loop to an if

switched the wile cmd2 not ins resp to be if cmd2 not in resp seems to be working.

* Update insteon_local.py

* import statement

Updated the import statement to import the instance of the insteon_local component not the hub Instance.

* updated import and the device assignment

update the import to import the instance of the insteon_local component not the hub.

* more changes to support the import change

* more changes to support the import change

* change to hass.data and add loop logic

* &&

* Update insteon_local.py

* Update insteon_local.py

* logic fixes and throttle

* reduce polling time

* brightness support

* import util

* hound fixes

* requirements file

* more hound fixes

* newline

* newline weirdness

* lint fixes

* more lint fixes

* switch state

* Update insteon_local.py

* log cmd2 for debugging

* assume success

* remove check for none

* fix comments

* fix comments again

* fix comments, add fixed version of lib, add support for timeout, add support for port, handle invalid login and connection problems

* fix logging exception

* fix hounceci-bot errors

* fix hounceci-bot errors

* requirements fix

* unique-id changes

* make dimmer off use saved ramp rate

* configurator working for lights

* configurator working for switches?

* configurator working for switches?

* include model names and fix lint errors

* lint fix

* fix exception order

* lint fixes

* fix lint errors

* update to use insteon local 0.38

* fix device id

* move status check to library

* move status check to library

* add SKU to setup

* lint fixes

* requirements

* linting

emlove committed:
Add SUPPORT_PLAY flag (#5181)

* Add SUPPORT_PLAY flag

* Add SUPPPORT_PLAY to existing media players

* Leave usage of new flag to device devs

Commit:
Update frontend

Commit:
Update frontend

Commit:
Update MDI

stjohnjohnson committed:
Fix #5188 by Closing the stream instead of Releasing it (#5235)

* Fix #5188 by Closing the stream instead of Releasing it

Closing just terminates the connection, release attempts to download all the contents before closing.  Since the MJPEG stream is infinite, it loads and loads content until the python script runs out of memory.

Source: https://github.com/KeepSafe/aiohttp/blob/50b1d30f4128abd895532022c726e97993987c7c/aiohttp/client_reqrep.py#L668-L672

* Update mjpeg.py

terrycarlin committed:
Update insteon_local.py (#5236)

Correct typo

michaelarnauts committed:
Add Lannouncer tts notify component (#5187)

* Add Lannouncer notify component

* Send message by opening a raw TCP socket instead of using requests. Cleanup of method validation.

* Use 'return' instead of 'return None'

Duoxilian commented:
@balloob: Please advise what you want me to do.

P.S. I am planning to make further updates to the ecobee code, as I found that when putting the thermostat into vacation mode confusing information shows up in hass. Not because something is wrong, but because you need to know that you are in vacation mode to make sense of this information. So I will need to add support for vacation mode and probably reflect the invalid states of other data somehow. I assume vacation mode is something that more thermostats will offer, but I guess I will start at the ecobee level and when others add this to other climate  components we can worry about generalizing.

pvizeli committed:
[device_traker/upc] New UPC connect box platform (#5100)

emlove committed:
Build libcec for Docker image (#5230)

* Build libcec for Docker image

* Update development dockerfile as well

* Dynamically load python paths for current version

fabaff committed:
Upgrade TwitterAPI to 2.4.3 (#5244)

kellerza committed:
[sensor/sma] SMA Solar Inverter sensor (#5118)

* Initial

* Rebase ensure_list

* timedelta & remove prints

fabaff committed:
Update pytz to 2016.10 (#5247)

kellerza committed:
[recorder] Include & Exclude domain fix & unit tests (#5213)

* Tests & domain fix

* incl/excl combined

andrey-git committed:
Expose supported_features of mqtt_json (#5250)

* Expose supported_features of mqtt_json

* Remove whitespace

glance- committed:
Fix async_volume_up / async_volume_down (#5249)

async_volume_up / async_volume_down should be async versions of
volume_up / volume_down, not a async version of the default variants of
volume_up / volume_down.

The previous code always called into the mediaplayers set_volume_level,
and never into volume_up / volume_down.

Signed-off-by: Anton Lundin <glance@acc.umu.se>

sander76 committed:
adding a default icon "blind" to a PowerView blinds scene. (#5210)

* adding a default icon "blind" to a PowerView blinds scene.

* Adding icon property to define blind icon. Removed it from the state attributes dict.

* fixing lint error

valentinalexeev committed:
Use SHA hash to make token harder to guess (#5258)

* Use SHA hash to make token harder to guess

Use hashlib SHA256 to encode object id instead of using it directly.

* Cache access token

Instead of generating a token on the fly cache it in the constructor.

* Fix lint

pvizeli committed:
Bugfix async device_tracker see callback (#5259)

joopert committed:
Add support for NAD receivers (#5191)

* Add support for NAD receivers

* remove self.update() in various methods

* remove setting attributes in various methods

* Change import to hass style

tchellomello committed:
Bump flux_led version and make use of PyPi package (#5267)

* Bump flux_led version and make use of PyPi package

* Makes script/gen_requirements_all.py happy

robbiet480 committed:
Merge pull request #5117 from gopalkildoliya/dev

Notify component for Facebook Messenger

Danielhiversen committed:
Add last triggered to script (#5261)

* Add last triggered to script

* Add tests for script last_triggered

pvizeli committed:
Cleanup language support on TTS (#5255)

* Cleanup language support on TTS

* change to default_language & address comments

* Cleanup not needed code / comment from paulus

nikdoof committed:
Fix TCP sensor to correctly use value_template (#5211)

* Fix TCP sensor to correctly use value_template

* Fix TCP component tests

* Update tcp.py

balloob committed:
Upgrade to aiohttp 1.2 (#4964)

* Upgrade to aiohttp 1.2

* Clean up emulated_hue tests

emlove committed:
Convert flic to synchronous platform. (#5276)

* Convert flic to synchronous platform.

pyflic is a synchronous library

* Move pyflic event loop to dedicated thread.

pavoni committed:
Bump pywemo version.

pavoni committed:
Merge pull request #5280 from home-assistant/bump-pywemo-version

Bump pywemo version.

pavoni committed:
Don't build Adafruit_BBIO - doesn't work on all platforms. (#5281)

* Don't build Adafruit_BBIO - doesn't work on all platforms.

* Disable pylint import warning on BBIO.

pvizeli commented:
Option 1 is the way to go :+1:

bachp committed:
Upgrade distro to 1.0.2 (#5291)

bachp committed:
Upgrade to voluptuous to 0.9.3 (#5288)

Commit:
Support for TrackR device trackers (#5010)

* Support for TrackR device trackers

* Change small style for hass

Duoxilian committed:
Rebased

Duoxilian committed:
Fixed lint error

pvizeli commented:
You PR is corrupt. Please don't merge: https://home-assistant.io/developers/development_catching_up/
I think you should make a new branch.

Duoxilian commented:
I did a rebase and then a big mess resulted in this pull. The code tests fine on my end. I will close this request and redo this change. I will perform it at the ecobee level, not the component level, so that the suggested change does not affect other thermostats.

Duoxilian commented:
This work will be continued in #5325
